LayoutTests/http/wpt/mediasession/toggleCapture.html

```diff
@@ -0,0 +1,146 @@
+<html>
+<body>
+<script src="../../resources/testharness.js"></script>
+<script src="../../resources/testharnessreport.js"></script>
+<iframe src="resources/toggleCapture-iframe.html" id=frame1></iframe>
+<iframe src="resources/toggleCapture-iframe.html" id=frame2></iframe>
+<script>
+
+promise_test(async () => {
+    return Promise.all([
+        new Promise(resolve => frame1.onload = resolve),
+        new Promise(resolve => frame2.onload = resolve),
+    ]);
+}, "Setup tests");
+
+promise_test(async (test) => {
+    if (!window.internals)
+        return;
+
+    let actionCount = 0;
+    navigator.mediaSession.setActionHandler("togglecamera", () => actionCount++);
+    navigator.mediaSession.setActionHandler("togglemicrophone", () => actionCount++);
+    navigator.mediaSession.setActionHandler("togglescreenshare", () => actionCount++);
+
+    await Promise.all([
+        frame1.contentWindow.startCapture(true, true),
+        frame2.contentWindow.startCapture(true, false)
+    ]);
+
+    let resultsPromise = Promise.all([
+        frame1.contentWindow.waitForToggleAction("togglecamera"),
+        frame1.contentWindow.waitForToggleAction("togglemicrophone"),
+        frame2.contentWindow.waitForNoToggleAction("togglecamera"),
+        frame2.contentWindow.waitForToggleAction("togglemicrophone"),
+    ]);
+    internals.setPageMuted("capturedevices");
+    assert_array_equals(await resultsPromise, [false, false, null, false]);
+
+    assert_equals(actionCount, 0);
+
+    resultsPromise = Promise.all([
+        frame1.contentWindow.waitForToggleAction("togglecamera"),
+        frame1.contentWindow.waitForToggleAction("togglemicrophone"),
+        frame2.contentWindow.waitForNoToggleAction("togglecamera"),
+        frame2.contentWindow.waitForToggleAction("togglemicrophone"),
+    ]);
+    internals.setPageMuted("");
+    assert_array_equals(await resultsPromise, [true, true, null, true]);
+
+    assert_equals(actionCount, 0);
+
+   frame1.contentWindow.stopCapture();
+   frame2.contentWindow.stopCapture();
+}, "togglecamera and togglemicrophone on various iframes");
+
+async function testActionEventOrder(track, action)
+{
+    let isActionHandled = false;
```

Comment:
can we move the test for `window.internals` here?

Reply:
Done